classes/UpgradeSelfCheck.php

```diff
@@ -226,11 +300,38 @@ public function getConfigDir()
      */
     public function getMaxExecutionTime()
     {
-        return $this->maxExecutionTime;
+        if (null !== $this->maxExecutionTime) {
+            return $this->maxExecutionTime;
+        }
+
+        return $this->maxExecutionTime = $this->checkMaxExecutionTime();
     }
 
+    /**
+     * @return bool
+     */
+    public function isPhpUpgradeRequired()
+    {
+        if (1 === (int) Configuration::get('PS_AUTOUP_IGNORE_PHP_UPGRADE')) {
+            return false;
+        }
+
+        if (null !== $this->phpUpgradeNoticelink) {
+            return $this->phpUpgradeNoticelink;
+        }
+
+        return $this->phpUpgradeNoticelink = $this->checkPhpVersionNeedsUpgrade();
```

Comment:
seems like `phpUpgradeNoticelink` can only be a boolean not a string, you should update its type doc And rename it to avoid confusion

Reply:
That's true, I forgot to update the property. Thanks